src/libarena/lib.rs

```diff
@@ -154,24 +155,54 @@ impl<T> TypedArena<T> {
         }
     }
 
+    #[inline]
+    pub fn alloc_slice(&self, slice: &[T]) -> &mut [T]
+        where T: Copy {
+        assert!(mem::size_of::<T>() != 0);
+        if slice.len() == 0 {
+            return unsafe { slice::from_raw_parts_mut(heap::EMPTY as *mut T, 0) };
```

Comment:
`&mut []` doesn't have a guaranteed stable address across instantiations and we depend on pointer comparisons working. That said, this would only *ever* be invoked once so maybe it's fine to do that.